### DISCUSSION TITLE
use (multi-)touchscreen to interact with multiple widgets

### DISCUSSION THREAD

**Discussion by olafklingt** [emilk/egui#398] [Q&A]
I am thinking about using egui to create a gui that is controlled with a touchscreen. Unlike your multi-touch example in the demo I don't want to use it to control with multiple fingers one object (the arrow) but to control with multiple fingers multiple widgets like buttons. Where should I look at to get an idea to realize such a thing? 
I am on linux.

**Comment by emilk** [comment]
This is not currently possible with egui. Letting multiple fingers interact with multiple widgets at once will be a lot of effort to implement.

**Reply by olafklingt** [reply]
this is where the raw_input touch events are transformed into gestures, right?
[https://github.com/emilk/egui/blob/0b52813f62de5a46dfde4c891c5f2f169b0aa8e1/egui/src/input_state/touch_state.rs](https://github.com/emilk/egui/blob/0b52813f62de5a46dfde4c891c5f2f169b0aa8e1/egui/src/input_state/touch_state.rs)
I would have to implement something different for my use-case.
If I would transform raw_input touch events into (for example) mouse-button-clicks at different positions could this be handled easily by multiple button widgets?
